### DISCUSSION TITLE
Wildfly Error (Unable to start a service)

### DISCUSSION THREAD

**Discussion by dalibor-rada** [Keyfactor/ejbca-ce#528] [Q&A] [answered]
Hi,

I'm trying to install EJBCA on my PC (well its not exactly my PC, I am using fresh Ubuntu 22.04.4 LTS as virtual machine), but every time I restart the server after installation I get these two errors.

I use:
openjdk-11jdk
mariadb-server
latest ejbca version

![a](https://github.com/Keyfactor/ejbca-ce/assets/162028714/d88063da-f974-4116-a51d-17102790ebfe)
![f](https://github.com/Keyfactor/ejbca-ce/assets/162028714/bad44900-82cc-4e96-b967-8659b0b39446)

EDIT: The server.log says ERROR [[org.jboss.as](http://org.jboss.as/)] (Controller Boot Thread) WFLYSRV0026: WildFly Full 26.0.0.Final (Wildfly COre 18.0.0.Final) started (with errors) in 5039ms - Started 338 of 610 services (29 services failed or missing dependencies, 341 services are lazy, passive or on-demand). I mean, it looks like it started somehow, but the adminweb is not working (Unable to connect).
Mainly I don’t understand the second one. I have that driver, because I followed exactly the instructions on the website (https://doc.primekey.com/ejbca/ejbca-installation/application-servers/wildfly-26). Obviously I’ve changed the database.properties and set the right database type and so on. I really don't know what to do. So could I please ask for help?

Thanks a lot

**Comment by primetomas** [comment, marked as answer]
Your error tells that the MariaDB driver is not istalled correctly.
Please verify this section performed correctly. It says that mariadb-java-client.jar is missing.
https://doc.primekey.com/ejbca/ejbca-installation/application-servers/wildfly-26#WildFly26-AddDatabaseDriver
